### PR TITLE
Peer testing unit tests and minor peer test changes

### LINKED ISSUE #55: Testing Strategies
**Is this related to a problem? Please describe.**

Need to explore unit testing strategies that can be implemented into a node js  project. For  unit testing check out jest and for API testing check out postman. 

**Describe the solution you'd like**

Include research into markdown file for the final report. 

**What aspects are unclear - what needs to be done, or how it should be accomplished?**
Please describe - if you cannot describe, this should likely be a Task.

**Additional context**

### LINKED ISSUE #78: Typescript Exploration  [Exploration]
**Is this related to a problem? Please describe.**

Every member on the team should get familiar with typescript syntax and concepts. 

**Additional context**

Here is some good documentation to get started: 

https://www.typescriptlang.org/docs/

### PR DESCRIPTION
**Pull request by edouarde1**
- Error handling functions work better now! 
-All unit tests pass
-Some changes to instructions for peer testing
-Learned how to make promises 

related to issues: #119 #107 
closes #78 
closes #55

### CODE REVIEW

**Comment by LoganParker** [comment]
"learned how to make promises"
I like this

**Review by vnajack — CHANGES_REQUESTED** [review]
Great teamwork!! ~~Is it possible to ignore `package-lock.json`? Once that's ignored,~~ I'll approve right away 😄

**Comment by LoganParker** [comment]
> Great teamwork!! Is it possible to ignore `package-lock.json`? Once that's ignored, I'll approve right away 😄

DO NOT - that will cause errors

**Comment by vnajack** [comment]
~~but is it ok with the +11,000 changes?~~ Logn educated me 😄

**Review by vnajack — APPROVED** [review]

File: speaking-portal-project/src/index.ts [hunk 1]
```diff
@@ -31,15 +31,15 @@ const recognizer_options: string[] = ['English (U.S.)', 'French', 'Japanese','En
 
 // Main Function
 async function main() {
-    // TODO: Change language_selection based on desired language
+    // TODO: Change case number during peer testing
     // English = 0, French = 1, Japanese = 2
```
Comment (LoganParker) [inline speaking-portal-project/src/index.ts]:
```suggestion
    // English = 0, French = 1, Japanese = 2, Wrong file = 3, Corrupted file = 4, Unique files = 5|6|7|8
```

File: speaking-portal-project/src/test/test.ts [hunk 2]
```diff
@@ -1,20 +1,37 @@
-import {test,expect} from '@jest/globals'
-import { doesFileExist,isFileReadable,isWavFile,isTextFileValid } from "./file_check";
+import {test,expect,describe} from '@jest/globals'
+import { doesFileExist,isWavFile,isTextFileValid, isWavFileValid, isTextEmpty } from "./file_check";
+import {} from 'fs'
```
Comment (LoganParker) [inline speaking-portal-project/src/test/test.ts]: empty import?
```suggestion
```

**Review by LoganParker — APPROVED** [review]
All looks great! Thanks for getting the testing fixed :)

File: speaking-portal-project/src/index.ts (hunk 1 above)
Comment (edouarde1) [reply speaking-portal-project/src/index.ts]: Does case 3 have a pointer to an empty string ?

File: speaking-portal-project/src/test/test.ts (hunk 2 above)
Comment (edouarde1) [reply speaking-portal-project/src/test/test.ts]: yes! Good catch

**Pull request merged**